### PR TITLE
Adds Gitter API authentication support

### PR DESCRIPTION
**Pull request by abishekvashok**
The Gitter API follows Oauth 2.0 for authenticating users. The flow is
based on browser redirects and code exchange as noted in Oauth 2.0. In
addition to this, `OkHttp` has been used as a networking library for its
ease of use and flexibility. `Butterknife` has also been implemented to reduce
code complexity.

A `WebView` client has been implemented which supports `javascript` and
loads the authentication page of Gitter, where Gitter prompts the user
to authenticate his/her self. After successful authentication, Gitter
redirects the `WebView` to a a webpage as indicated by `redirect uri`
specified during creation of the app on Gitter with a code parameter
whose value is stored and is sent along with other parameters such as
client_id etc, as a post request using `OkHttp` to a token generation page
of Gitter which cross checks the parameters and returns an access code.
The access code is stored in shared preferences for future uses. **The
`OkHttp` libraries `async` calls have been used with the method `enqueue`
as `synchronous` calls should run on `Async Tasks` which leaks memory.**

The current Chat Activity has been deleted and has been replaced by a
Dashboard Activity which is redirected to by another starting activity
known as `SplashActivity` which is used for redirects. When no internet
and access_code is available Splash Activity redirects to
`NoNetwork`Activity`.

In addition to the above, the style colors of the app has also been
changed.

**Fixes issue:** #1 

CC: @garg-anuj @the-dagger @championswimmer

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -24,8 +24,9 @@ dependencies {
     androidTestImplementation('com.android.support.test.espresso:espresso-core:2.2.2', {
         exclude group: 'com.android.support', module: 'support-annotations'
     })
-    implementation 'com.android.support:appcompat-v7:25.3.1'
+    implementation 'com.android.support:appcompat-v7:25.4.0'
     testImplementation 'junit:junit:4.12'
     implementation 'com.android.support.constraint:constraint-layout:1.0.2'
-    implementation 'com.android.support:design:25.3.1'
+    implementation 'com.android.support:design:25.4.0'
+    compile 'com.squareup.okhttp3:okhttp:3.8.0'
```
Comment (garg-anuj) [inline app/build.gradle]: We should rather use retrofit. @the-dagger What do you day?

File: app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -10,15 +13,19 @@
         android:supportsRtl="true"
         android:theme="@style/AppTheme">
         <activity
-            android:name=".ChatActivity"
+            android:name=".SplashActivity"
```
Comment (garg-anuj) [inline app/src/main/AndroidManifest.xml]: Splash activity is loosing its trend in market. Rather use home screen as screen with no/less network usage and easy to load content.

**Comment by garg-anuj** [comment]
@abishekvashok Its high time to start learning [Butterknife](http://jakewharton.github.io/butterknife/)

File: app/src/main/AndroidManifest.xml (hunk 2 above)
Comment (abishekvashok) [inline app/src/main/AndroidManifest.xml]: Well it redirects the user. Its use will be visible only in further implementations.

Comment (abishekvashok) [inline app/src/main/AndroidManifest.xml]: Well it redirects the user. Its use will be visible only in further implementations.

File: app/build.gradle (hunk 1 above)
Comment (abishekvashok) [inline app/build.gradle]: But I can see that okhttp 3 is more better than retrofit. Isnt it so?

**Comment by abishekvashok** [comment]
I would implement butterknife.

File: app/build.gradle (hunk 1 above)
Comment (garg-anuj) [inline app/build.gradle]: Nope. It is not true. Under the hood retrofit also use okhttp 3 It just reduce the boiler plate code for call and data conversion.

Comment (the-dagger) [inline app/build.gradle]: OKHttp is good for now, but depending upon the use case, we might want to shift to retrofit later on.

Comment (garg-anuj) [inline app/build.gradle]: Okay the. As @the-dagger says. Continue with with OKHttp 3

Comment (abishekvashok) [inline app/build.gradle]: Ok :)

**Comment by abishekvashok** [comment]
@garg-anuj @the-dagger  implemented `Butterknife` and rebased the commits. :)

**Comment by abishekvashok** [comment]
Things done so far:
- [ ] Take out the current chat activity as it is a boiler plate only.
- [ ] Create a new Gitter app in order to make use of the gitter api
- [ ] Implement authentication as mentioned in a complex process by the Gitter API at: https://developer.gitter.im/docs/authentication
- [ ] For the last phase a networking client can be used as the stock way sucks like OkHttp
- [ ] Store the generated access code in shared preference and display it in dashboard activity.
- [ ] When no access_code is there make the user to authenticate again and if internet is also absent in addition prompt the user to authenticate.
- [ ] Add Butterknife

**Pull request merged**